lib/utils.js

```diff
@@ -12,21 +16,61 @@ var basicAuth = require('basic-auth');
  * @param   {string}   password Expected password
  * @returns {function} Express 4 middleware requiring the given credentials
  */
-exports.basicAuth = function(username, password) {
-	return function(req, res, next) {
-
-		if (!username || !password) {
-			console.log('Username or password is not set.');
-			return res.send('<h1>Error:</h1><p>Username or password not set. <a href="https://github.com/alphagov/govuk_prototype_kit/blob/master/docs/deploying.md#3-set-a-username-and-password">See guidance for setting these</a>.</p>');
-		}
-		
-		var user = basicAuth(req);
-
-		if (!user || user.name !== username || user.pass !== password) {
-			res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
-			return res.sendStatus(401);
-		}
-
-		next();
-	};
+function basicAuth (username, password) {
+  return function(req, res, next) {
+
+    if (!username || !password) {
+      console.log('Username or password is not set.');
+      return res.send('<h1>Error:</h1><p>Username or password not set. <a href="https://github.com/alphagov/govuk_prototype_kit/blob/master/docs/deploying.md#3-set-a-username-and-password">See guidance for setting these</a>.</p>');
+    }
+
+    var user = basicAuth(req);
+
+    if (!user || user.name !== username || user.pass !== password) {
+      res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
+      return res.sendStatus(401);
+    }
+
+    next();
+  };
+}
+
+// Checks all modules in package.json are available, quit if not available
+function checkNodeModules(){
+  var nodeModules = require(__dirname + '/../package.json');
+  for (var nodeModule in nodeModules.dependencies) {
+    checkNodeModule(nodeModule);
+  }
+};
+
+// Check a single node module, quit app if not available
+function checkNodeModule(moduleName){
+  if(isModuleAvailableSync(moduleName) !== true ){
+    console.error('\nERROR: ' + moduleName + ' module missing. Try running `npm install`\n');
+    process.exit(0);
+  }
+};
+
+// checks if a given node module is available to load
+function isModuleAvailableSync (moduleName){
+  var ret = false; // return value, boolean
```

Comment:
Try to avoid comments on the same line. It's easier to read them when on a separate line and allows changes to code/comments to be seen separately in git diffs.